### ISSUE TITLE
False positive for ST11 when using fields implicitly via wildcard select

### ISSUE THREAD

**Issue by dobsontom** [sqlfluff/sqlfluff#6511]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running `sqlfluff lint` on a specific model in dbt project raises a false positive warning for rule ST11:

```
L:  22 | P:  15 | ST11 | Joined table 'sourcings' not referenced elsewhere in
                       | query [structure.unused_join]
```

### Expected Behaviour

SQLFluff should not raise an ST11 warning for this model as fields from this table are used (albeit implicitly via `SELECT *`).

### Observed Behaviour

`sqlfluff lint` raises the following warning when tested both with and without the `EXCLUDE` keyword:

```
L:  22 | P:  15 | ST11 | Joined table 'sourcings' not referenced elsewhere in
                       | query [structure.unused_join]
```

### How to reproduce

Run `sqlfluff lint` on the following dbt model:

```
with simulation_source_data_reference as (

    select * from {{ ref('stg_simulation__simulation_source_data_reference') }}

),

sourcings as (

    select * from {{ ref('fct_simulation_sourcings') }}

),

ref_table as (

    select

        simulation_source_data_reference.*,
        sourcings.* exclude sourcing_job_id

    from simulation_source_data_reference

    left join sourcings
        on simulation_source_data_reference.sourcing_job_id = sourcings.sourcing_job_id

)

select * from ref_table
```

### Dialect

Snowflake, templater dbt.

### Version

`sqlfluff`, version 3.3.0
`sqlfluff-templater-dbt`, version 3.3.0
Python, version 3.12.8

### Configuration

```
[sqlfluff]

dialect = snowflake
templater = dbt

exclude_rules = AM06, LT05, RF02, RF03, RF04, RF06, ST06, ST07

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by dobsontom** [comment]
Bumping this as it is still an issue.

**Comment by FrankTub** [comment]
I'm running into the same issue. Sample part of when it occurs:

```sql
with
kyc_dossiers as (
    select *
    from {{ ref('beehive_kyc_dossiers') }}
),

unpivot_kyc_dossiers as (
    select
        kds.*,
        cast(fpids.value as integer) as party_id
    from kyc_dossiers as kds,
        lateral flatten(input => kds.party_ids) as fpids
)
```

It complains about:

```
Joined table 'kyc_dossiers as kds' not referenced elsewhere in querysqlfluff(ST11: structure.unused_join)
```
